src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java

```diff
@@ -272,17 +458,143 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
- * To configure the check to produce a violation on a class with too many fields:
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *   public void foo() { // ok
+ *       System.out.println("Hello");
+ *   }
+ *   public void boo() {
+ *       ; // violation
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to produce a violation on a class or interface with too many fields:
  * </p>
  * <pre>
  * &lt;module name=&quot;DescendantToken&quot;&gt;
- *   &lt;property name=&quot;tokens&quot; value=&quot;CLASS_DEF&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;CLASS_DEF,INTERFACE_DEF&quot;/&gt;
  *   &lt;property name=&quot;limitedTokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
  *   &lt;property name=&quot;maximumDepth&quot; value=&quot;2&quot;/&gt;
- *   &lt;property name=&quot;maximumNumber&quot; value=&quot;10&quot;/&gt;
+ *   &lt;property name=&quot;maximumNumber&quot; value=&quot;1&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Foo { // ok
```

Comment:
All examples should be compilable, please fix this example to be compilable. You can do something like:
```
class A {...}
class B {...}
interface C {...}
interface D {...}

```

Reply:
Done.

Reply:
Please prove by CLI that your update will compile.

Reply:
In fact, there are a few other examples that will not compile.

@romani we missed `javac` of all examples in gist

Reply:
In fact, there are a few other examples that will not compile.

@romani we missed `javac` of all examples in gist

@coderdeadpool all code should be compilable, even if we have snippet (not full file)

Reply:
@nrmancuso for this example, the cli o/p is
```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="DescendantToken">
        <property name="tokens" value="CLASS_DEF,INTERFACE_DEF"/>
        <property name="limitedTokens" value="VARIABLE_DEF"/>
        <property name="maximumDepth" value="2"/>
        <property name="maximumNumber" value="1"/>
      </module>
  </module>
</module>
```
```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat test1.java
public class A { // ok
    private int field1;

    // Some code
  }
  public class B { // violation
    private int field1;
    private int field2;

    // Some code
  }
  public interface C { // ok
    int FIELD_1 = 1;
  }
  public interface D { // violation
    int FIELD_1 = 1;
    int FIELD_2 = 2;
  }
```
```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ java -jar checkstyle-10.9.3-SNAPSHOT-all.jar -c config1.xml test1.java
Starting audit...
[ERROR] C:\Users\hp\Desktop\test1.java:6:3: Count of 2 for 'CLASS_DEF' descendant 'VARIABLE_DEF' exceeds maximum count 1. [DescendantToken]
[ERROR] C:\Users\hp\Desktop\test1.java:15:3: Count of 2 for 'INTERFACE_DEF' descendant 'VARIABLE_DEF' exceeds maximum count 1. [DescendantToken]
Audit done.
Checkstyle ends with 2 errors.
```

Reply:
@coderdeadpool compilation requires javac, please try `javac test1.java`

Reply:
@nrmancuso I ran ```javac test1.java``` and got
```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ javac test1.java
test1.java:1: error: class A is public, should be declared in a file named A.java
public class A { // ok
       ^
test1.java:6: error: class B is public, should be declared in a file named B.java
  public class B { // violation
         ^
test1.java:12: error: interface C is public, should be declared in a file named C.java
  public interface C { // ok
         ^
test1.java:15: error: interface D is public, should be declared in a file named D.java
  public interface D { // violation
         ^
4 errors
```

Reply:
and same for all the other examples

Reply:
Yes, this means that code is not able to be compiled and is not valid Java code. Make sure all examples are able to be compiled. Checkstyle demands that all code is compilable by javac. You can see more about restrictions/ limitations of checkstyle at https://checkstyle.sourceforge.io/writingchecks.html#Limitations

Reply:
@nrmancuso Done.

Reply:
All examples are now compilable